src/main/java/org/jenkinsci/plugins/github/config/GitHubServerConfig.java

```diff
@@ -59,22 +62,45 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(GitHubServerConfig.class);
 
     /**
+     * Common prefixes that we should remove when inferring a {@link #name}.
+     *
+     * @since 1.27.0
+     */
+    private static final String[] COMMON_PREFIX_HOSTNAMES = {
+        "git.",
+        "github.",
+        "vcs.",
+        "scm.",
+        "source."
+    };
```

Comment:
Because if somebody provides an url like `https://github.example.com` we really would rather give that the name `example` rather than `github example` because the `github` is redundant information... on the other hand if it was `https://github1.example.com` then we want `github1 example` as the suggested display name